cgi-bin/LJ/Sysban.pm

```diff
@@ -209,6 +209,12 @@ sub sysban_check {
         return 0 unless scalar @domains >= 2;
         my $domain = "$domains[-2].$domains[-1]";
         return 1 if $check->('email_domain', $domain);
+        # also check for three-element domain bans
```

Comment:
I agree that there is probably a more elegant solution to be found here, but I don't think replacing a simple split expression with a complicated regular expression is a good idea for comprehensibility and maintainability.  Also, this won't work for all cases - if a user gives us a four element subdomain, and we've implemented a sysban based on the last three elements, I believe this check will still fail.